src/mutate-errors-to-todos.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  TodoData,
+  todoFilePathFor,
+  _buildTodoDatum,
+} from '@ember-template-lint/todo-utils';
+import type { ESLint, Linter } from 'eslint';
+import { TodoResultMessage } from 'types/estlint-type-extension';
+
+/**
+ * Mutates all errors present in the todo dir to todos in the results array.
+ * @param results ESLint results array
+ */
+export async function mutateTodoErrorsToTodos(
+  results: ESLint.LintResult[],
+  todoMap: Map<string, TodoData>
+): Promise<void> {
+  results.forEach((result) => {
+    (result.messages as TodoResultMessage[]).forEach((message) => {
+      if (message.severity !== 2) {
+        return;
+      }
+
+      // we only mutate errors that are present in the todo map, so check if it's there first
+      const todoDatum = _buildTodoDatum(result, message as Linter.LintMessage);
```

Comment:
I think if you type the arrow function param, you don't need to cast here (unless you're doing this on purpose).

Reply:
Yeah I'm doing this on purpose because `message` here is actually a `TodoResultMessage` where I changed the severity to support `-1` but I need to pass it to `_buildTodoDatum` as a `Linter.LintMessage`.